### PR TITLE
Kernel#extend raises a TypeError when the argument is not a Module

### PR DESCRIPTION
**Pull request by fazibear**

### CODE REVIEW

File: core/kernel/extend_spec.rb [hunk 1]
```diff
@@ -69,5 +69,9 @@ class C
     it "raises a RuntimeError" do
       lambda { @frozen.extend @module }.should raise_error(RuntimeError)
     end
+
+    it "raises a TypeError when the argument is not a Module" do
+      lambda { KernelSpecs::M.send(:extend, Class.new) }.should raise_error(TypeError)
+    end
```
Comment (eregon) [inline core/kernel/extend_spec.rb]: This should not be in the `describe "on frozen instance" do` but just above :wink: 
The receiver could also just be some mock like `o = mock('o')`.
`extend` is not private so it does not need a `send`.

**Comment by fazibear** [comment]
Fixed :)

**Comment by eregon** [comment]
@fazibear Thanks for your contribution!
I squashed your commits and added a last little nitpick :smile:

**Comment by eregon** [comment]
(Closed by 7551ce69c7ad8b0d395200d23cf492378d93c1ae)

**Pull request closed without merging**